ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -231,17 +235,29 @@ private class SqlLedger(
               parties.toSet[String]
         }
 
-      LedgerEntry.Transaction(
-        tx.commandId,
-        transactionId,
-        tx.applicationId,
-        tx.submitter,
-        tx.workflowId,
-        tx.ledgerEffectiveTime,
-        timeProvider.getCurrentTime,
-        mappedTx,
-        mappedDisclosure
-      )
+      val recordTime = timeProvider.getCurrentTime
+      if (recordTime.isAfter(tx.maximumRecordTime)) {
+        LedgerEntry.Rejection(
+          recordTime,
+          tx.commandId,
+          tx.applicationId,
+          tx.submitter,
+          RejectionReason.TimedOut(
+            s"RecordTime $recordTime is after MaximumRecordTime ${tx.maximumRecordTime}")
+        )
+      } else {
+        LedgerEntry.Transaction(
+          tx.commandId,
+          transactionId,
+          tx.applicationId,
+          tx.submitter,
+          tx.workflowId,
+          tx.ledgerEffectiveTime,
+          recordTime,
```

Comment:
it's not your fault, but I just noticed that we have a race in the persistence pipeline. As the entries can be persisted concurrently, they might get saved with non strictly increasing record times. The question is: When `offsetT2 > offsetT1 ` is/must `recordTimeT2 >= recordTimeT1` implied from it?

Reply:
I take it back. I forgot that it's actually called via a callback, where the order is guaranteed.